app/controllers/ConsignmentReferencesController.scala

```diff
@@ -44,12 +44,12 @@ class ConsignmentReferencesController @Inject()(
   def displayPage(): Action[AnyContent] = (authenticate andThen ileQueryFeatureDisabled andThen getJourney(JourneyType.ARRIVE, JourneyType.DEPART)) {
     implicit request =>
       val references = request.answersAs[MovementAnswers].consignmentReferences
-      Ok(consignmentReferencesPage(references.fold(form())(form().fill(_))))
+      Ok(consignmentReferencesPage(references.fold(form(request.answers.`type`))(form(request.answers.`type`).fill(_))))
   }
 
   def saveConsignmentReferences(): Action[AnyContent] =
     (authenticate andThen ileQueryFeatureDisabled andThen getJourney(JourneyType.ARRIVE, JourneyType.DEPART)).async { implicit request =>
-      form()
+      form(request.answers.`type`)
```

Comment:
Worth something like
```
val frm = form(request.answers.`type`)
```
as class field?

Reply:
request.answers.`type` is a dynamic value that can change with each call to the saveConsignmentReferences method, so I'm not sure how you could make this an immutable class field val?

Reply:
:+1: